activerecord/lib/active_record/persistence.rb

```diff
@@ -55,6 +55,79 @@ def create!(attributes = nil, &block)
         end
       end
 
+      # Inserts a record (or multiple records) into the database. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved models and it does not
+      # trigger Active Record callbacks or validations. However, values passed to #insert_many will still go through
+      # Active Record's normal type casting and serialization.
+      #
+      # The +attributes+ parameter can be either a Hash or an Array of Hashes. These Hashes describe the
+      # attributes on the objects that are to be created.
+      #
+      # ==== Options
+      #
+      # [:returning]
+      #   (Postgres-only) An array of column names that should be returned for all successfully inserted rows.
+      #   By default, the primary key(s) will be returned for every inserted rows. Can be set to `false` to return nothing.
+      #
+      # [:on_conflict]
+      #   A hash that describes how the database should respond to duplicate rows:
+      #
+      #     { do: :nothing } — skip duplicate rows
+      #
+      #     { do: :update } — update duplicate rows (perform an UPSERT)
```

Comment:
Can you make sure the docs note which one is the default?

Reply:
🤔 Good question...

Right now, if you don't pass `on_conflict: { anything } ` to `insert_many`, it just won't include the `ON CONFLICT` clause and the `INSERT` will fail if it violates any unique constraints (it'll raise `ActiveRecord::RecordNotUnique`).

If you pass `on_conflict: { do: anything_other_than_nothing_or_update }`, it'll raise an `ArgumentError` and ask for `:nothing` or `:update`.

What do you think? Should I just clarify this in the docs? Or also add an explicit `on_conflict: { do: :raise }` option that is equivalent to `on_conflict: false`?

It might be nice to also support `on_conflict: :skip` and `on_conflict: :raise` as shorthand for `on_conflict: { do: :nothing }` and `on_conflict: { do: :raise }`. Do you think that's worth the little extra work / processing to add now?

(The inner hash in `on_conflict: { do: :nothing }` bugs me a little; but you _might_ want to do `on_conflict: { do: :nothing, columns: %w{ author_id name } }` if your unique constaint is on something other than the primary key.)